### PR TITLE
4 crud food

### PR DESCRIPTION
**Pull request by jVucetic**

### CODE REVIEW

File: src/main/java/com/g1dra/foodmanager/services/FoodService.java [hunk 1]
```diff
@@ -16,8 +21,11 @@
 @RequiredArgsConstructor
 @Transactional
 public class FoodService {
+    @Autowired
```
Comment (d-vucetic) [inline src/main/java/com/g1dra/foodmanager/services/FoodService.java:24]: Ovo autwired je nepotrebno cini mi se. On ce sam odraditi load te klase. Malo je i meni zbrka ovdje koji je najbolji nacin. Ali ova fasada je dobra stvar svaka cast

**Comment by jVucetic** [comment]
1. Constructor level is most recommended way
 
    Proper use of IoC. We define the object dependencies at the constructor level, so we can’t miss any of them.
 
    Easy for testing- You can create mocks from your context instances in unit tests and inject them via the constructor
    
2. Setter Level
    
    Recommended for *optional* dependencies - This is because you won’t be safe from constructing the desired object without one or more dependencies
    
3. Field injection using @Autowired 
    
    Testing becomes a headache.
    
    By default, Spring resolves *@Autowired* entries by type. If more than one bean of the same type is available in the container, the framework will throw a fatal exception.
[Article about autowiring](https://dzone.com/articles/explain-major-benefits-amp-limits-of-auto-wiring-i)

**Comment by d-vucetic** [comment]
A koje je objasnjenje kako zna da ubaci ove sto nemaju ni anotaciju autowired ?

**Comment by d-vucetic** [comment]
Jes sad se i ja sjetih da mora all args. Hvala :+1:

**Pull request merged**